.github/workflows/causal_lm_cpp.yml

```diff
@@ -681,36 +681,42 @@ jobs:
+      matrix: {runs-on: [ubuntu-20.04-16-cores, macos-13]}
+    runs-on: ${{ matrix.runs-on }}
     steps:
       - uses: actions/checkout@v4
-        with:
-          submodules: recursive
+        with: {submodules: recursive}
       - uses: actions/setup-python@v4
-        with:
-          python-version: 3.11
-      - name: Install OpenVINO
-        run: |
-          mkdir ./ov/
+        with: {python-version: 3.12}
+      - run: mkdir ./ov/
+      - if: ${{ 'ubuntu-20.04-16-cores' == matrix.runs-on }}
+        run: >
           curl ${{ env.l_ov_link }} | tar --directory ./ov/ --strip-components 1 -xz
-          sudo ./ov/install_dependencies/install_openvino_dependencies.sh
-      - name: Build app
-        run: |
-          source ./ov/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
+          && sudo ./ov/install_dependencies/install_openvino_dependencies.sh
+      - if: ${{ 'macos-13' == matrix.runs-on }}
+        run: >
+          curl ${{ env.m_ov_link }} | tar --directory ./ov/ --strip-components 1 -xz
+          && brew install coreutils scons
+      - run: OpenVINO_DIR=./ov/runtime/cmake/ cmake -DCMAKE_BUILD_TYPE=Release -B ./build/ ./
+      - run: >
+          LD_LIBRARY_PATH=${{ github.workspace }}/ov/runtime/3rdparty/tbb/lib/:$LD_LIBRARY_PATH
           cmake --build ./build/ --config Release --target visual_language_chat -j
-      - name: Download and convert a model and an image
-        run: |
+      - run: >
           source ./ov/setupvars.sh
-          python -m pip install ./thirdparty/openvino_tokenizers/[transformers] --pre --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
-          python -m pip install --upgrade-strategy eager -r ./samples/requirements.txt --pre --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
-          python ./samples/cpp/visual_language_chat/export_MiniCPM-V-2_6.py ./miniCPM-V-2_6/
-          wget https://github.com/openvinotoolkit/openvino_notebooks/assets/29454499/d5fbbd1a-d484-415c-88cb-9986625b7b11
-      - run: |
+          && python -m pip install --upgrade-strategy eager ./thirdparty/openvino_tokenizers/[transformers] -r ./samples/requirements.txt --pre --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
+      - run: >
+          source ./ov/setupvars.sh
+          && python ./samples/cpp/visual_language_chat/export_MiniCPM-V-2_6.py ./miniCPM-V-2_6/
+      - run: wget https://github.com/openvinotoolkit/openvino_notebooks/assets/29454499/d5fbbd1a-d484-415c-88cb-9986625b7b11
+      - run: >
           source ./ov/setupvars.sh
-          timeout 2m ./build/samples/cpp/visual_language_chat/visual_language_chat ./miniCPM-V-2_6/ d5fbbd1a-d484-415c-88cb-9986625b7b11
+          && ./build/samples/cpp/visual_language_chat/visual_language_chat ./miniCPM-V-2_6/ d5fbbd1a-d484-415c-88cb-9986625b7b11
           <<< $'What is on the image?\nWhat is special on the image?'
+        timeout-minutes: 110
```

Comment:
The value is in testing the new use case on mac.
Reducing the test to single prompt would reduce the time by 50 minutes. But I don't think it's enough because there's also export that also takes 50 minutes. Should I delete the mac test?

Reply:
IMO, we need to optimize CI as now it takes significant time.. but this tests will make situation ever worse

Reply:
Removed